pyo3-macros-backend/src/method.rs

```diff
@@ -17,19 +17,107 @@ use crate::{
 };
 
 #[derive(Clone, Debug)]
-pub struct FnArg<'a> {
+pub struct RegularArg<'a> {
     pub name: &'a syn::Ident,
     pub ty: &'a syn::Type,
-    pub optional: Option<&'a syn::Type>,
-    pub default: Option<syn::Expr>,
-    pub py: bool,
     pub attrs: PyFunctionArgPyO3Attributes,
-    pub is_varargs: bool,
-    pub is_kwargs: bool,
-    pub is_cancel_handle: bool,
+    pub default_value: Option<syn::Expr>,
+    pub option_wrapped_type: Option<&'a syn::Type>,
+}
+
+#[derive(Clone, Debug)]
+pub struct VarArg<'a> {
+    pub name: &'a syn::Ident,
+    pub ty: &'a syn::Type,
+}
+
+#[derive(Clone, Debug)]
+pub struct KwArg<'a> {
+    pub name: &'a syn::Ident,
+    pub ty: &'a syn::Type,
+}
```

Comment:
Leaning into the `args`, `kwargs` names, what do you think of these as alternatives? `KwArg` in particular makes me think it's a keyword-only single argument, not `**kwargs`.

```suggestion
#[derive(Clone, Debug)]
pub struct ArgsArg<'a> {
    pub name: &'a syn::Ident,
    pub ty: &'a syn::Type,
}

#[derive(Clone, Debug)]
pub struct KwargsArg<'a> {
    pub name: &'a syn::Ident,
    pub ty: &'a syn::Type,
}
```

Reply:
Yeah, i can see your point here. `KwargsArg` i think is fine, but `ArgsArg` does not really read clearer to me, maybe `VarargsArg` plus a small doc comment on both?

Reply:
Yep that's good with me 👍